WalletWasabi/Wallets/WalletManager.cs

```diff
@@ -33,8 +33,23 @@ public WalletManager(Network network, WalletDirectories walletDirectories)
 				WalletDirectories = Guard.NotNull(nameof(walletDirectories), walletDirectories);
 				Wallets = new Dictionary<Wallet, HashSet<uint256>>();
 				Lock = new object();
-				AddRemoveLock = new AsyncLock();
+				StartStopWalletLock = new AsyncLock();
 				CancelAllInitialization = new CancellationTokenSource();
+
+				if (WalletDirectories is { })
+				{
+					foreach (var fileInfo in WalletDirectories.EnumerateWalletFiles())
+					{
+						try
+						{
+							AddWallet(fileInfo.Name);
```

Comment:
This could be done as follow. In that case you don't need to call the RegisterServices on the wallet. 
```suggestion
							AddWallet(fileInfo.Name, this);
```

Reply:
I guess you are talking about this, so my answer is the same here: https://github.com/zkSNACKs/WalletWasabi/pull/3305#discussion_r396075364